src/netbase.cpp

```diff
@@ -445,6 +439,19 @@ bool static ConnectSocketDirectly(const CService &addrConnect, SOCKET& hSocketRe
     if (!SetSocketNonBlocking(hSocket, true))
         return error("ConnectSocketDirectly: Setting socket to non-blocking failed, error %s\n", NetworkErrorString(WSAGetLastError()));
```

Comment:
Will fix __func__.

SetSocketNonBlocking should take care of closing the socket, but an explicit one here wouldn't hurt either.

CloseSocket is really annoying. I'll RAII it directly after this PR.